src/FieldWarning/Assets/UI/Ingame/UnitLabel/WaypointOverlayBehavior.cs

```diff
@@ -11,66 +11,68 @@
  * the License for the specific language governing permissions and limitations under the License.
  */
 
+using System.Linq;
 using PFW.Units;
 using PFW.Units.Component.Movement;
-using System.Collections.Generic;
+using PFW.Units.Component.OrderQueue;
 using UnityEngine;
 
-public class WaypointOverlayBehavior : MonoBehaviour
+namespace PFW.UI.Ingame.UnitLabel
 {
-
-    private PlatoonBehaviour _platoon;
-    
-    private LineRenderer _lineR;
-
-    public void Destroy()
+    public class WaypointOverlayBehavior : MonoBehaviour
     {
-        Destroy(gameObject);
-    }
+        private PlatoonBehaviour _platoon;
+        private LineRenderer _lineR;
 
-    // Update is called once per frame
-    private void Update()
-    {
-        if (_platoon.ActiveWaypoint == null)
+        public void Destroy()
         {
-            // TODO: should prob just set inactive...
-            _lineR.gameObject.SetActive(false);
-            return;
+            Destroy(gameObject);
         }
 
-        _lineR.gameObject.SetActive(true);
+        private void Update()
+        {
+            var activeOrder = _platoon.OrderQueue.ActiveOrder;
+            if (activeOrder == null)
+            {
+                // TODO: should prob just set inactive...
+                _lineR.gameObject.SetActive(false);
+                return;
+            }
 
-        // +2 for the active waypoint and our self
-        _lineR.positionCount = _platoon.Waypoints.Count + 2;
 
-        _lineR.SetPosition(0, _platoon.transform.position);
+            _lineR.gameObject.SetActive(true);
 
-        // destination is normally dequeued so we need to get this separately from
-        // the rest of the waypoints
-        _lineR.SetPosition(1, _platoon.ActiveWaypoint.Destination);
+            // +2 for the active waypoint and our self
+            _lineR.positionCount = _platoon.OrderQueue.Orders.Count() + 2;
 
+            _lineR.SetPosition(0, _platoon.transform.position);
 
-        int idx = 0;
-        foreach (MoveWaypoint wp in _platoon.Waypoints)
-        {
-            // +2 for the destination and ourselves previously inserted into this line
-            _lineR.SetPosition(idx + 2, wp.Destination);
-            idx++;
+            // destination is normally dequeued so we need to get this separately from
+            // the rest of the waypoints
+            _lineR.SetPosition(1, activeOrder.Destination);
```

Comment:
One bug we have here is that fire pos orders are also showing up as a move preview. Doesn't have to be fixed instantly but its one more reason to have different queues.